From 5577aa6fa8e8122689a04a750e6680b595ac98e3 Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Mon, 06 Dec 2021 04:02:43 +0200
Subject: [PATCH] [SW-65065] gaudi2: secureboot: remove btl version

In secureboot flow there is no btl version. Version
string that preboot stores in SRAM also need not
have btl version as it causes mismatch with secureboot
version in fw image fit. This patch makes sure
btl version is written to version string only in
case of legacy flow.

Change-Id: I3bdd58be4836e4cc0fa912d4bc7dbcc66144675c
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index 5912446..a739925 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -138,32 +138,42 @@
 	return true;
 }
 
-static void preboot_version(void)
+/* Return size of btl version copied */
+static int copy_btl_version(void)
 {
-	unsigned long offset = (unsigned long) preboot_vers[PREBOOT_BTL_VER];
 	u32 btlver = regmap_readl(mmPREBOOT_VER_OFFSET);
-	int i, btl_ver_sz;
-
-	preboot_puts("\n");
-	if (hl_is_legacy_boot()) {
-		preboot_puts_raw(preboot_vers[PREBOOT_BTL_VER]);
-		preboot_puth(btlver, true);
-	}
-	preboot_puts_raw(preboot_ver_str);
-	preboot_puts("\n");
+	int i = 0;
 
 	/* get to the end of the predefined btl string */
 	for (i = 0; i < VERSION_MAX_LEN; i++)
 		if (preboot_vers[PREBOOT_BTL_VER][i] == '\0')
 			break;
-
 	/* convert and copy the btl version */
 	i += preboot_uint2hexstring(&preboot_vers[PREBOOT_BTL_VER][i],
 				    btlver, true);
 
 	/* separate the btl from preboot */
 	preboot_vers[PREBOOT_BTL_VER][i++] = '\n';
-	btl_ver_sz = i;
+
+	return i;
+}
+
+static void preboot_version(void)
+{
+	u32 btlver = regmap_readl(mmPREBOOT_VER_OFFSET);
+	unsigned long offset = (unsigned long) preboot_vers[PREBOOT_BTL_VER];
+	int i = 0, btl_ver_sz = 0;
+	bool is_legacy_boot = hl_is_legacy_boot();
+
+	preboot_puts("\n");
+	if (is_legacy_boot) {
+		preboot_puts_raw(preboot_vers[PREBOOT_BTL_VER]);
+		preboot_puth(btlver, true);
+		i = btl_ver_sz = copy_btl_version();
+	}
+	preboot_puts_raw(preboot_ver_str);
+	preboot_puts("\n");
+
 
 	/* copy the preboot version */
 	for (int j = 0; j < VERSION_MAX_LEN - btl_ver_sz - 1; j++) {
